### PR TITLE
12798 - can generate casting errors when recovering from failed piece build

### PR DESCRIPTION
**Pull request by Cattlesquat**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/widget/PieceSlot.java [hunk 1]
```diff
@@ -192,13 +192,14 @@ protected void clearExpandedPiece() {
    */
   public GamePiece getPiece() {
     if (c == null && pieceDefinition != null) {
-      final AddPiece comm =
-        (AddPiece) GameModule.getGameModule().decode(pieceDefinition);
-      if (comm == null) {
+      final Command cp = GameModule.getGameModule().decode(pieceDefinition); 
+      if ((cp == null) || cp.isNull() || !(cp instanceof AddPiece)) {
         System.err.println("Couldn't build piece " + pieceDefinition);
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/widget/PieceSlot.java:197]: Uh wow, we're printing the message to stderr?! That should go the Chatter, like Bad Data Errors do.

**Comment by Cattlesquat** [comment]
Yes this one is on my list to re-address once PR #184 is merged, since it has all the cool new Bad Module Data code.

**Comment by Cattlesquat** [comment]
Okay, this was apparently a high-traffic block of shitty code! Had to resolve conflicts with other fixes from the other branch! But anyway unless I made a typo this should be good to go.

**Comment by Cattlesquat** [comment]
(er, well, right now travis won't build anything, but I don't think it's actually me)

**Comment by Cattlesquat** [comment]
Okay now it's ready

**Pull request merged**